### PR TITLE
lnd connection checks

### PR DESCRIPTION
**Pull request by joostjager**
Make sure we've always checked the lnd instance for a correct configuration and add a version check.

### CODE REVIEW

**Review by Torakushi — APPROVED** [review]
LGTM

File: lnd/client.go [hunk 1]
```diff
@@ -72,38 +80,80 @@ func NewLndClient(ctx context.Context, cfg Config) (LndClient, error) {
...
+	}
+
+	// Request node info.
+	info, err := l.lnClient.GetInfo(ctx, &lnrpc.GetInfoRequest{})
+	if err != nil {
+		return err
+	}
+
+	// Clip off unofficial commit indication from version reported by lnd.
+	versionString := info.Version
```
Comment (Torakushi) [inline lnd/client.go]: Why not use verrpc? (so we don't depend on a particular string format)

Comment (joostjager) [reply lnd/client.go]: I can use verrpc, but then need to replicate the version comparison logic (major > major, major == major, major < major, minor ..., patch ...).

An in-between solution could be to use verrpc in combination with go-version for the comparison?

Comment (joostjager) [reply lnd/client.go]: Went for this one.

Comment (Torakushi) [reply lnd/client.go]: Ah yeah... Just checked.
Indeed it will work,  I would prefer that instead of relying to the lnd string format!

Comment (Torakushi) [reply lnd/client.go]: Really prefer that! good job

**Pull request merged**